document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/mapper/smallfull/model/ixbrl/period/Period.java

```diff
@@ -0,0 +1,150 @@
+package uk.gov.companieshouse.document.generator.accounts.mapper.smallfull.model.ixbrl.period;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.gson.Gson;
+
+import java.util.Objects;
+
+public class Period {
+
+    @JsonProperty("current_period_start_on")
+    private String currentPeriodStartOn;
+
+    @JsonProperty("current_period_end_on")
+    private String currentPeriodEndsOn;
+
+    @JsonProperty("previous_period_start_on")
+    private String previousPeriodStartOn;
+
+    @JsonProperty("previous_period_end_on")
+    private String previousPeriodEndsOn;
+
+    @JsonProperty("current_period_start_on_formatted")
+    private String currentPeriodStartOnFormatted;
+
+    @JsonProperty("current_period_end_on_formatted")
+    private String currentPeriodEndOnFormatted;
+
+    @JsonProperty("previous_period_start_on_formatted")
+    private String previousPeriodStartOnFormatted;
+
+    @JsonProperty("previous_period_end_on_formatted")
```

Comment:
we could remove this field as it does not exist in the small-full-accounts.html template yet. This can be added when it is needed

Reply:
I've kept it included as we have the current and should give the option for the previous. It's there if someone needs it in the future.